implementations/rust/ockam/ockam_command/src/enroll/command.rs

```diff
@@ -463,12 +463,32 @@ async fn get_user_space(
         "Marked {} as your default Space, on this machine.\n",
         color_primary(space.name.clone())
     ))?;
-
-    opts.terminal.write_line(fmt_log!("This Space does not have a Subscription attached to it."))?
-        .write_line(fmt_log!("As a courtesy, we created a temporary Space for you, so you can continue to build.\n"))?
-        .write_line(fmt_log!("Please subscribe to an Ockam plan within two weeks {}", color_uri("https://www.ockam.io/pricing")))?
-        .write_line(fmt_log!("{}\n", "If you don't subscribe in that time, your Space and all Projects will be permanently deleted.".color(OckamColor::FmtWARNBackground.color())))?;
-
+    if let Some(subscription) = &space.subscription {
+        opts.terminal.write_line(fmt_log!(
+            "This Space has a {} Subscription attached to it.",
+            color_primary(&subscription.name)
+        ))?;
+        if subscription.is_free_trial {
+            if is_new {
+                opts.terminal.write_line("")?
+                    .write_line(fmt_log!("As a courtesy, we created a temporary Space for you, so you can continue to build.\n"))?
+                    .write_line(fmt_log!("Please subscribe to an Ockam plan within two weeks {}", color_uri("https://www.ockam.io/pricing")))?
+                    .write_line(fmt_log!("{}", "If you don't subscribe in that time, your Space and all Projects will be permanently deleted.".color(OckamColor::FmtWARNBackground.color())))?;
+            } else if let (Some(start_date), Some(end_date)) =
+                (&subscription.start_date, &subscription.end_date)
+            {
+                opts.terminal.write_line("")?
+                    .write_line(fmt_log!("Your free trial started on {} and will end on {}.\n", start_date, end_date))?
+                    .write_line(fmt_log!("Please subscribe to an Ockam plan before the trial ends to avoid any service interruptions {}", color_uri("https://www.ockam.io/pricing")))?
+                    .write_line(fmt_log!("{}", "If you don't subscribe in that time, your Space and all Projects will be permanently deleted.".color(OckamColor::FmtWARNBackground.color())))?;
+            }
+        }
+    } else {
+        opts.terminal.write_line(fmt_log!(
+            "This Space does not have a Subscription attached to it."
```

Comment:
Any consequence? Should we tell the user how much cooler it would be to have a subscription attached to it?

Reply:
> Any consequence?
For now, I'm not sure if this code is reachable, as the user should have at least the "developer-free" subscription. This is still in the works on the elixir side, so we might want to wait until that's ready.